### PR TITLE
[SWAP] Integrate SPL Token extra calls for AppExchange

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Create client services lib:
 * CAL
 * Trust

Add new APDU to BOLOS for PKI transmission.

### ❓ Context

- **JIRA or GitHub link**: [LIVE-14938]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14938]: https://ledgerhq.atlassian.net/browse/LIVE-14938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/exchange/swap/completeExchange.ts [hunk 1]
```diff
@@ -242,6 +261,20 @@ const completeExchange = (
...
+
+async function sendPKI(transport: Transport) {
+  // FIXME: version number hardcoded
+  const { descriptor, signature } = await calService.getCertificate(
+    transport.deviceModel!.id,
+    "1.3.0",
+  );
+
+  await loadPKI(transport, "TRUSTED_NAME", descriptor, signature);
+}
```
Comment (Justkant) [inline libs/ledger-live-common/src/exchange/swap/completeExchange.ts:276]: Do we want to fix the missing model id and version number before merging this ?

File: libs/ledgerjs/packages/hw-bolos/src/loadPKI.integ.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    // Given
+    const descriptor =
+      "0101010201023501043601021004010500001302000314010120086e66745f6d657461300200053101043201213401013321035e6c1020c14dc46442fe89f97c0b68cdb15976dc24f24c316e7b30fe4e8cc76b";
+    const signature =
+      "30440220774aa1a064c9bd5ff2c820156b4af106d464d8ca0eb2de5f2956c8b52356496202205ef40dd54beb3c1534d011af9f581f426141d39338b3da6abf4023254cbef72c";
+
+    // When & Then
+    await expect(loadPKI(transport, "TRUSTED_NAME", descriptor, signature)).rejects.not.toThrow();
+    await (() => new Promise(f => setTimeout(f, 1000)))();
+  });
```
Comment (Justkant) [inline libs/ledgerjs/packages/hw-bolos/src/loadPKI.integ.test.ts:26]: This test part does work at the moment right ?

File: libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  TRUSTED_NAME: 0x04,
+  BACKUP_PROVIDER: 0x05,
+  RECOVER_ORCHESTRATOR: 0x06,
+  PLUGIN_METADATA: 0x07,
+  COIN_META: 0x08,
+  SEED_ID_AUTH: 0x09,
+  UNKNOWN: 0x0a,
+};
+
+const OkStatus = 0x9000;
```
Comment (Justkant) [inline libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts:22]: Should we use the code from the transport package ?

**Review by Justkant — DISMISSED** [review]
LGTM, thx for creating all those packages

File: libs/ledgerjs/packages/hw-bolos/src/loadPKI.integ.test.ts (hunk 2 above)
Comment (sprohaszka-ledger) [reply libs/ledgerjs/packages/hw-bolos/src/loadPKI.integ.test.ts:26]: Yes

File: libs/ledger-live-common/src/exchange/swap/completeExchange.ts (hunk 1 above)
Comment (sprohaszka-ledger) [reply libs/ledger-live-common/src/exchange/swap/completeExchange.ts:276]: For your need, I can comment this function to quickly merge.

File: libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts (hunk 3 above)
Comment (sprohaszka-ledger) [reply libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts:22]: I don't see it in the Transport package :/

File: libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    INS,
+    keyUsage[key],
+    P2,
+    Buffer.concat([
+      new Uint8Array(descriptorBuffer),
+      new Uint8Array(Buffer.from("15", "hex")),
+      new Uint8Array(Buffer.from([signatureBuffer.length])),
+      new Uint8Array(signatureBuffer),
+    ]),
+    handledErrorCode.concat(StatusCodes.OK),
```
Comment (Justkant) [inline libs/ledgerjs/packages/hw-bolos/src/loadPKI.ts:45]: small detail but we could concat that outside the function

**Review by Justkant — APPROVED** [review]

**Pull request merged**